server/sonar-server/src/main/java/org/sonar/server/computation/filemove/SourceSimilarityImpl.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.filemove;
+
+import java.util.List;
+
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+// TODO evaluate https://commons.apache.org/sandbox/commons-text/jacoco/org.apache.commons.text.similarity/CosineSimilarity.java.html
+// TODO another possible algorithm : just take the intersection of line hashes. That would allow to support block moving.
+public class SourceSimilarityImpl implements SourceSimilarity {
+
+  // TODO verify algorithm http://stackoverflow.com/questions/6087281/similarity-score-levenshtein
+  @Override
+  public <T extends Object> int score(List<T> left, List<T> right) {
+    int distance = levenshteinDistance(left, right);
+    return (int) (100 * (1.0 - ((double) distance) / (max(left.size(), right.size()))));
+  }
+
+  // TODO verify https://commons.apache.org/sandbox/commons-text/jacoco/org.apache.commons.text.similarity/LevenshteinDistance.java.html
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)